### PR TITLE
add option to wait on GH release status

### PR DESCRIPTION
**Pull request by evansde77**
@petevg @appeltel @ksnavely @shudgston 

Adding a pause to wait for CI to run on the release branch during upload

### CODE REVIEW

**Comment by ksnavely** [comment]
LGTM, nice and easy way to work with CI on protected branches.

Can we make the timeout configurable though? For those groggy Travis mornings.

File: src/cirrus/github_tools.py [hunk 1]
```diff
@@ -10,9 +10,39 @@
...
+
+    """
+    config = load_configuration()
+    token = get_github_auth()[1]
+    url = "https://api.github.com/repos/{org}/{repo}/commits/{branch}/status".format(
+        org=config.organisation_name(),
+        repo=config.package_name(),
+        branch=branch_name
+    )
+    print url
```
Comment (petevg) [inline src/cirrus/github_tools.py]: Stray print statement?

Comment (evansde77) [inline src/cirrus/github_tools.py]: yup :)

**Comment by evansde77** [comment]
Timeout is configurable via cirrus.conf or CLI option

File: src/cirrus/release.py [hunk 2]
```diff
@@ -329,12 +344,45 @@ def _trigger_jenkins_release(config, new_version, level):
...
+    """
+    time_spent = 0
+    status = branch_status(branch_name)
+    while status == 'pending':
+        if time_spent > timeout:
+            LOGGER.error("Exceeded timeout for branch status {}".format(branch_name))
+            break
+        status = branch_status(branch_name)
+        time.sleep(2)
+        time_spent += 2
```
Comment (appeltel) [inline src/cirrus/release.py]: Can we make this configurable as well? For projects where the CI check is normally on the order of minutes 2 seconds seems overly frequent

**Comment by ksnavely** [comment]
+1 cool

File: src/cirrus/release.py (hunk 2 above)
Comment (evansde77) [inline src/cirrus/release.py]: yup

File: src/cirrus/release.py [hunk 3]
```diff
@@ -329,12 +344,45 @@ def _trigger_jenkins_release(config, new_version, level):
...
+    Wait for CI checks to complete for the branch named
+    """
+    time_spent = 0
+    status = branch_status(branch_name)
+    while status == 'pending':
+        if time_spent > timeout:
+            LOGGER.error("Exceeded timeout for branch status {}".format(branch_name))
+            break
+        status = branch_status(branch_name)
+        time.sleep(2)
```
Comment (petevg) [inline src/cirrus/release.py]: Does git have a feed that we can listen on for status changes? Ideally, I'd to not have to wait an extra two seconds for something to complete if I don't need to. Impatient dev is impatient :-)

Comment (petevg) [inline src/cirrus/release.py]: ... that said, this is +1 for an initial release. Listening on the feed could be an improvement down the line ...

File: src/cirrus/release.py (hunk 2 above)
Comment (ksnavely) [inline src/cirrus/release.py]: What about just using timeout/20 or something?

**Comment by petevg** [comment]
+1, but with desire for less polling, and more long polling/streaming if possible.

**Comment by appeltel** [comment]
+1

File: src/cirrus/release.py (hunk 3 above)
Comment (evansde77) [inline src/cirrus/release.py]: for now, grabbing the status in a dumb polling loop should work (especially as it is jenkins that sits and waits for it) I didnt see an obvious API call for it, maybe something with a webhook would work...

**Pull request merged**